### PR TITLE
SLCORE-436 add vulnerabilityProbability to Issue

### PR DESCRIPTION
**Pull request by sophio-japharidze-sonarsource**

### CODE REVIEW

File: analysis-engine/src/test/java/org/sonarsource/sonarlint/core/analysis/sonarapi/DefaultFilterableIssueTests.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 import org.sonarsource.sonarlint.core.analysis.api.Issue;
 import org.sonarsource.sonarlint.core.analysis.container.analysis.filesystem.DefaultTextPointer;
 import org.sonarsource.sonarlint.core.analysis.container.analysis.filesystem.DefaultTextRange;
+import org.sonarsource.sonarlint.core.commons.VulnerabilityProbability;
```
Comment (damien-urruty-sonarsource) [inline analysis-engine/src/test/java/org/sonarsource/sonarlint/core/analysis/sonarapi/DefaultFilterableIssueTests.java:31]: Dead import ?

File: server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/hotspot/ServerHotspot.java [hunk 2]
```diff
@@ -31,20 +33,23 @@ public class ServerHotspot {
...
-    String filePath,
-    TextRange textRange,
-    Instant creationDate,
-    boolean resolved) {
+                       String ruleKey,
+                       String message,
+                       String filePath,
+                       TextRange textRange,
+                       Instant creationDate,
+                       boolean resolved, VulnerabilityProbability vulnerabilityProbability) {
```
Comment (damien-urruty-sonarsource) [inline server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/hotspot/ServerHotspot.java:44]: Indentation too ?

File: server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/XodusServerIssueStore.java [hunk 3]
```diff
@@ -231,14 +233,15 @@ private static ServerHotspot adaptHotspot(Entity storedHotspot) {
...
+    var vulnerabilityProbability = VulnerabilityProbability.valueOf((String) storedHotspot.getProperty(VULNERABILITY_PROBABILITY_PROPERTY_NAME));
     return new ServerHotspot(
       (String) requireNonNull(storedHotspot.getProperty(KEY_PROPERTY_NAME)),
       (String) requireNonNull(storedHotspot.getProperty(RULE_KEY_PROPERTY_NAME)),
       requireNonNull(storedHotspot.getBlobString(MESSAGE_BLOB_NAME)),
       filePath,
       textRange,
       (Instant) requireNonNull(storedHotspot.getProperty(CREATION_DATE_PROPERTY_NAME)),
-      Boolean.TRUE.equals(storedHotspot.getProperty(RESOLVED_PROPERTY_NAME)));
+      Boolean.TRUE.equals(storedHotspot.getProperty(RESOLVED_PROPERTY_NAME)), requireNonNull(vulnerabilityProbability));
```
Comment (damien-urruty-sonarsource) [inline server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/XodusServerIssueStore.java:244]: I don't think you need the requireNonNull here, the `valueOf` could throw earlier so it's not possible. And I don't think it's needed to catch on the valueOf, I don't see this case happening

**Review by damien-urruty-sonarsource — APPROVED** [review]
Some comments but overall LGTM :+1:

**Comment by sophio-japharidze-sonarsource** [comment]
Covered in #554

**Pull request closed without merging**